spec/ics-25-handler-interface/README.md

```diff
@@ -0,0 +1,314 @@
+}
+```
+
+`ClientOptions` contains all the parameter choices required to create a client.
+
+```golang
+type ClientOptions struct {
+  ClientKind  kind
+  RootOfTrust rootOfTrust
+}
+```
+
+`ClientInfo` contains information about an existing client.
+
+```golang
+type ClientInfo struct {
+  ClientKind  kind
+  RootOfTrust rootOfTrust
+}
+```
+
+`createClient` creates a new client and returns an automatically allocated identifier.
+
+```coffeescript
+function createClient(ClientOptions options) -> string
+```
+
+`queryClient` queries a client by a known identifier, returning the associated metadata and root of trust if found.
+
+```coffeescript
+function queryClient(string id) -> Maybe<ClientInfo>
+```
+
+`updateClient` updates an existing client with a new header, returning an error if the client was not found or the header was not a valid update.
+
+```coffeescript
+function updateClient(string id, Header header) -> Maybe<Err>
+```
+
+Implementations of `createClient`, `queryClient`, and `updateClient` are defined in ICS 2.
+
+
+### Connections
+
+By default, connections are unowned, but closure is permissioned to the module which created the connection.
+
+`ConnectionKind` enumerates the connection types supported by the handler implementation.
+
+```golang
+type ConnectionKind enum {
+  Transit
+  Receive
+  Broadcast
+  Bidirectional
+}
+```
+
+`ConnectionInfo` contains metadata about & state of an existing connection.
+
+```golang
+type ConnectionInfo struct {
+  ConnectionOptions options
+  ConnectionState   state
+}
+```
+
+`initConnection` tries to create a new connection with the provided options, failing if the client is not found or the options are invalid.
+
+```coffeescript
+function initConnection(identifier, desiredVersion, desiredCounterpartyIdentifier, lightClientIdentifier) -> Maybe<err>
+```
+
+`tryConnection` tries to initialize a connection based on an initialization attempt on another chain (part one of the three-way connection handshake).
+
+```coffeescript
+function tryConnection(desiredIdentifier, counterpartyIdentifier, desiredVersion, counterpartyLightClientIdentifier, lightClientIdentifier, proofInit) -> Maybe<err>
+```
+
+`ackConnection` acknowledges a connection in progress on another chain (part two of the three-way connection handshake).
+
+```coffeescript
+function ackConnection(identifier, agreedVersion, proofTry)
+```
+
+`confirmConnection` finalizes a connection (part three of the three-way connection handshake).
+
+```coffeescript
+function confirmConnection(identifier, proofAck)
+```
+
+`queryConnection` queries an existing connection by known identifier, returning the associated metadata if found.
+
+```coffeescript
+function queryConnection(string id) -> Maybe<ConnectionInfo>
+```
+
+`initCloseConnection` initiates the graceful connection closing process. It will fail if there are any open channels using the connection.
+
+```coffeescript
+function initCloseConnection(string id) -> Maybe<Err>
+```
```

Comment:
I think there is a difference between '(string id)' and '(identifier)', not sure I understand it well, could you please clarify, maybe in some earlier section.

Reply:
No, sorry for the confusion. Changed to `string identifier` everywhere.

Identifiers must follow the ruleset as defined in ICS 24, added a link.